src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NullTestAroundInstanceOfCheck.java

```diff
@@ -0,0 +1,207 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+
+/**
+ * <p>
+ * Checks if unnecessary null check is used around instanceof expression.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code tokens} - tokens to check
+ * Default value is:
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_IF">
+ * LITERAL_IF</a>.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;NullTestAroundInstanceOf&quot;/&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * class Test {
+ *
+ *     if (myObj != null &amp;&amp; myObj instanceof Object) { // violation
+ *
+ *     }
+ *
+ *     if (myObj != null) { // violation
+ *         if (myObj instanceof Object) {
+ *
+ *         }
+ *     }
+ *
+ *    if ((myObj != null) &amp;&amp; (myObj instanceof Object)) {}  // violation
+ * }
+ * </pre>
+ *
+ * @since 8.34
+ */
+@StatelessCheck
+public class NullTestAroundInstanceOfCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message null.before.instanceof in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "null.test.around.instanceof";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getAcceptableTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return new int[] {
+            TokenTypes.LITERAL_IF,
+        };
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return CommonUtil.EMPTY_INT_ARRAY;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        final DetailAST landNode = ast.findFirstToken(TokenTypes.EXPR)
+                .findFirstToken(TokenTypes.LAND);
+        if (landNode == null) {
+            if (checkOnNextLine(ast)) {
+                log(ast, MSG_KEY);
+            }
+        }
+        else {
+            final DetailAST notEqualNode = landNode.findFirstToken(TokenTypes.NOT_EQUAL);
+            final DetailAST instanceOfNode =
+                    landNode.findFirstToken(TokenTypes.LITERAL_INSTANCEOF);
+            if (notEqualNode != null && instanceOfNode != null
```

Comment:
it seems like this code can be extracted to separate method and used inside `visitToken` and `checkOnNextLine`

In several places we perform similar validation

What do you think?

Reply:
I agree, done